### ISSUE TITLE
TickSliderItem allows removal when allowAdd == False

### ISSUE THREAD

**Issue by edbarnard** [pyqtgraph/pyqtgraph#744]
A `TickSliderItem` has a `allowAdd` optional initialization argument. When it is set to False, it is not possible to add additional ticks. However, the current behavior allows the user to remove the ticks, but does not allow the user to add them back. The correct behavior probably should be to disable removal when `allowAdd` is False.

**Comment by 2xB** [comment]
It is possible to prohibit removing a tick using `tick.removeAllowed = False`. Therefore prohibiting the removal of any tick would be performed setting `removeAllowed` to `False`, e.g. using `TickSliderItem.listTicks`.

It turns out that this mechanism is broken, so I fix it in PR #1440. 

Should `allowAdd` also default `removeAllowed` to `False` for new ticks? I have no strong opinion on this design decision, and would therefore pledge not to change it (every change is a potential cause for error for other users).
I would suggest to add a new `allowRemove` property that acts as the default of `tick.removeAllowed`, and will add a PR to open discussion on that.

In the meantime, here is an example code of what works with PR #1440:

```python3
import pyqtgraph as pg
from pyqtgraph.Qt import QtCore
import numpy as np

app = pg.mkQApp()

class CustomWidget(pg.GraphicsView):
    def __init__(self, parent=None, *args, **kargs):
        pg.GraphicsView.__init__(self, parent, useOpenGL=False, background=None)
        self.item = pg.TickSliderItem(*args, **kargs)

        for pos in (0, 0.5, 1):
            tick = self.item.addTick(pos)
            tick.removeAllowed = False # Possibility 1

        # Possibility 2
        for tick, value in self.item.listTicks():
            tick.removeAllowed = False

        self.setCentralItem(self.item)
        self.setFixedHeight(31)

w = CustomWidget()
w.show()

app.exec_()
```

**Comment by 2xB** [comment]
Update: I noticed that `removeAllowed` is not a regular property of `Tick`, but monkeypatched by `GradientEditorItem`. Therefore this was no bug. I updated my PR to #1441, making it a regular property of `Tick` and enabling its use with `TickSliderItem`.
